wallet/models.py

```diff
@@ -1,5 +1,9 @@
 from django.db import models
 from django.contrib.auth.models import User
+from django.db.models.signals import pre_save
+from django.dispatch import receiver
+from django.core.exceptions import ValidationError
+from django.utils import timezone
 
 
 class Category(models.Model):
```

Comment:
About the relationship between category and user, is there any other way to represent it? How does that compare to the current solution?

Also, do we have a requirement about the category hierarchy?

Reply:
1. About the relationship between category and user, user can only see the categories created by the system(which means those default categories, when a user just registered those categories will show) and categories created by themselves. User should not allow seeing other user's categories.
2. About category hierarchy, we can remove it now, it does make my work more complicated.